src/ff1.rs

```diff
@@ -38,15 +38,18 @@ fn generate_s<CIPH: BlockCipher>(ciph: &CIPH, r: &[u8], d: usize) -> Vec<u8> {
     let mut s = Vec::from(r);
     s.reserve(d);
     {
-        let mut j = 1;
+        let mut j = BigUint::one();
```

Comment:
It seems overkill to use a bigint here. In practice the number of blocks is unlikely to be more than 255. Can we just assert that this is the case?